### ISSUE TITLE
Put license text into source files

### ISSUE THREAD

**Issue by tobimensch** [docopt/docopt.cpp#45]
It's easier for applications to include your source files, when the license text is on top of every .h and .cpp file, because every future "user/developer" of the file will immediatly see what the license is. All you're seeing right now is the copyright, which even might scare a view people of. Of course there are the license files and the problem is you're using two licenses.

I would suggest putting the MIT license in every source file, which is short, and to also put one ore two lines mentioning the alternative Boost license underneath the MIT license. 

Why is this important? I quote the howto on the GPL for this (because they have good lawyers):

"Whichever license you plan to use, the process involves adding two elements to each source file of your program: a copyright notice (such as “Copyright 1999 Terry Jones”), and a statement of copying permission, saying that the program is distributed under the terms of the GNU General Public License (or the Lesser GPL)."

http://www.gnu.org/licenses/gpl-howto.en.html

**Comment by jpaulos** [comment]
Ditto.  I had the same journey of discovery:  fear that I couldn't use this because of the "scary" in-source copyright statement, then the hunt for separate license files, and finally commenting here to tell the tale.  Excited to put this to work now, though.

**Comment by ghisvail** [comment]
Indeed, if you take the source files alone, all you read if "All rights reserved" which gives no clue that the code is subject to open-source licensing terms. The minimum would be to add the following statement:
```
This program is free software: you can redistribute it and/or modify
it under the terms of the MIT and/or Boost licenses at your convenience.
See the LICENSE-MIT and LICENSE-Boost-1.0 files for details.
```

**Comment by Johnnynator** [comment]
Clear licensing terms would be really nice. Right now it looks like some random License file checked out without any hint to which files they might apply.
